### PR TITLE
Feat/movegroups

### LINKED ISSUE #111: Basic Arm Functionality

### PR DESCRIPTION
**Pull request by MKerner3**
# Description
This PR does the following:
- Adds move group planning to the arm for trajectory execution
- Adds a demo file to test path planning and execution within rviz
- Allows the robot to be configured with the move group c++ interface

Fixes #111

# Testing Steps
## Test Case 1
1. cd to your rover colcon
2. `colcon build`
3. `. install/setup.bash`
4. `ros2 launch urc_arm_moveit_config demo.launch.py`

Expectation: The robot arm loaded in rviz, with the ability to move it around and execute path plans.

# Design Doc
[Link to Design Doc](https://docs.google.com/document/d/1AJjhGS5n1R8NQjK8jW5OeVoDND41m0tU5d3CTdZHPOI/edit)

# Self Checklist
- [x] I have formatted my code using `ament_uncrustify --reformat`
- [x] I have tested that the new behavior works

### CODE REVIEW

File: urc_arm_moveit_config/config/WalliiArmV3.ros2_control.xacro [hunk 1]
```diff
@@ -5,10 +5,10 @@
 
         <ros2_control name="${name}" type="system">
             <hardware>
-                
+                <!-- TODO make a launch param to switch between real/fake hardware -->
```
Comment (harrislegobrick) [inline urc_arm_moveit_config/config/WalliiArmV3.ros2_control.xacro:8]: Needs a new issue (probably a good new member issue).

File: urc_gazebo/urdf/WalliiArmV3.urdf [hunk 2]
```diff
@@ -555,24 +555,16 @@
...
-  <!-- mock_components/GenericSystem -->
-  <!-- gazebo_ros2_control/GazeboSystem -->
- 
-
-    <ros2_control name="arm_control" type="system">
+  
+  <ros2_control name="arm_control" type="system">
     <hardware>
-      <plugin>gazebo_ros2_control/GazeboSystem</plugin>
+      <!-- TODO make a launch param to switch between real/fake hardware -->
```
Comment (harrislegobrick) [inline urc_gazebo/urdf/WalliiArmV3.urdf:561]: Should also be part of the new member issue above.

**Review by harrislegobrick — APPROVED** [review]
Writing this down as it may be exclusive to my setup but is still worth writing down.

The `moveit_ros_move_group` fails to run when launching the demo file. This is fixed by manually running the node in a new terminal with `ros2 run moveit_ros_move_group movegroup ...` with the `...` being specific launch parameters based on tmp files. The tmp file specifics are displayed in the terminal when running the demo file as an error.

**Review by davidcalderon03 — APPROVED** [review]
GigaChad

**Pull request merged**